### ISSUE TITLE
[PCLVisualizer] strange behavior spinOnce() member of PCLVisualizer class

### ISSUE THREAD

**Issue by AlfaLegion** [PointCloudLibrary/pcl#5195]
I have built PCL 1.12.1 from source code with support VTK 9.1. When I compile a simple code:
```
#include <pcl/visualization/pcl_visualizer.h>
int main()
{
	pcl::visualization::PCLVisualizer::Ptr viewer (new pcl::visualization::PCLVisualizer ("3D Viewer"));	
  	viewer->setBackgroundColor (0, 0, 0);
  	viewer->addCoordinateSystem (1.0);
  	pcl::PointXYZ a(1,1,1);
	pcl::PointXYZ b(0,0,-1);
  	viewer->addLine(a,b);
  	viewer->initCameraParameters ();
  	
    	while (!viewer->wasStopped()) {
        	viewer->spinOnce(100);
    	}
    return 1;
}
```
Description: 
When I launched the program, the window appeared and immediately disappeared. When I started debugging, I noticed that the spinOnce () method had gone into waiting.

**Comment by larshg** [comment]
Which OS?

I have tried your example and it works fine with vtk 9.0.3, and I don't have vtk 9.1 yet, as I use whats available on VCPKG.

**Comment by AlfaLegion** [comment]
Ubuntu 20.04
Sometimes an error message appears in the console: I ran my example with psl 1.10 installed via apt-same result. 
X connection to ��GfCV broken (explicit kill or server shutdown).
X connection to  broken (explicit kill or server shutdown).

I do not know what to do.

**Comment by larshg** [comment]
Could sound like a driver issue with X or maybe the GPU - what graphic card do you use?

**Comment by AlfaLegion** [comment]
Nvidia RTX3090

**Comment by michhelle132** [comment]
Ubuntu 18.04
After updating to VTK 9.1 and PCL 1.12, I have immediate segmentation fault after spinOnce()

**Comment by AlfaLegion** [comment]
Which c++ standard are you using  to build PCL and VTK?

**Comment by michhelle132** [comment]
PCL uses 14, as it is set in the PCL CmakeLists.
VTK I can not see anything about setting the standard, so it uses the default compiler version which is 11. I can try to build VTK with 14, if it may help.

**Comment by AlfaLegion** [comment]
Strangely, I ran the example above and it did work with PCL 1.10

**Comment by AlfaLegion** [comment]
I compiled VTK with c++14 and built PCL. Result is the same.

**Comment by larshg** [comment]
Do you use nvidia's proprietary drivers? What if you make a simple VTK application - does that work? Like this:https://kitware.github.io/vtk-examples/site/Cxx/GeometricObjects/CylinderExample/

**Comment by AlfaLegion** [comment]
Yes, I use nvidia's proprietary drivers-470  Yes, I make a simple VTK application and it works.

**Comment by dripdripdr0p** [comment]
Hello,

I had similar issue with VTK 9.1 and PCL 1.12.
spinOnce() never worked, the error was "X11::Xcursor not found; custom cursors will be ignored".
Spin() did work. Simple VTK application above also did work.

I downgraded VTK to 7.1 and PCL to 1.11.1 and now it works

**Comment by yasamoka** [comment]
I believe this has been resolved by #5252 and can be closed.
